Libs/vtkSegmentationCore/vtkOrientedImageDataResample.cxx

```diff
@@ -1835,9 +1835,125 @@ bool vtkOrientedImageDataResample::IsLabelInMask(
   return valueFound;
 }
 
+//----------------------------------------------------------------------------
+bool vtkOrientedImageDataResample::IsSegmentationScalarTypeValid(vtkImageData* image)
```

Comment:
It is a bit complex that the scalar types are checked in two slightly different ways (one is segmentation-specific as it excludes long int and float types, the other is somewhat-segmentation-specific as it excludes long int types).

It would be better to have just one segmentation-specific check, which based on the scalar typa and range tells if the the image type is TYPE_OK, TYPE_CONVERSION_NEEDED, TYPE_CONVERSION_TRUNCATION_NEEDED, TYPE_ERROR. Depending on this, warning or error is reported and/or logged and casting is called. Casting does not need to do any warnings (it is a casting function, so the developer must know that casting may cause data loss; but maybe in the future if we need it then we could add an enum return value to distinguish between lossless and potentially lossy conversion).